src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -66,5 +69,11 @@ public class ProjectRestoreMetadata
         /// True if $(TargetFrameworks) is used and the build is using Cross Targeting.
         /// </summary>
         public bool CrossTargeting { get; set; }
+
+        /// <summary>
+        /// Whether or not to restore the packages directory using the legacy format, which write original case paths
+        /// instead of lowercase.
+        /// </summary>
+        public bool RestoreLegacyPackagesDirectory { get; set; }
```

Comment:
Add support to the PackageSpec reader/writer also, otherwise it is really easy to drop the value